lib/engine_socketio.js

```diff
@@ -180,52 +181,55 @@ SocketIoEngine.prototype.step = function (requestSpec, ee) {
   };
 
   function preStep(context, callback){
-    var namespace = requestSpec.emit.namespace || "/";
-    self.getSocket(namespace, context, function(err, socket){
+    // Set default namespace in emit action
+    requestSpec.emit.namespace = requestSpec.emit.namespace || "/";
+
+    self.loadContextSocket(requestSpec.emit.namespace, context, function(err, socket){
       if(err) {
         return callback(err, context);
       }
 
-      return f(context, callback, socket);
+      return f(context, callback);
     });
   }
 
   if(requestSpec.emit) {
+
```

Comment:
prestep behaves as a decorator for function `f`. It ensures that the socket for the emit namespace is loaded before executing `f`. 

Omitting it would add unnecessary complexity inside the `f` function. Separation  keeps everything a bit more in line with the single responsibility principle. I ensured that the invocation parameters of `preStep` and `f` are the same.